frontend/views/containers/chatroom/file-attachment/attachments-utils.js

```diff
@@ -0,0 +1,58 @@
+export const blobToDataUrl = (blob: Blob): Promise<any> => {
+  // serialize blob to base64
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader()
+    reader.onloadend = () => resolve(reader.result) // e.g. "data:video/mpeg;base64,..."
+    reader.onerror = reject
+    reader.readAsDataURL(blob)
+  })
+}
+
+export const dataUrlToBlob = (dataURL: string): any => {
+  const [meta, data] = dataURL.split(',')
+  const mimeType = meta?.match(/:(.*?);/)?.[1]
+
+  if (!mimeType || !data) {
+    console.error('Error converting base64 to blob: mimeType or base64 is missing')
+    return null
+  }
+
+  const byteString = atob(data)
```

Comment:
There's another issue here that I didn't comment on (assuming you'd use `fetch`), but this is incorrect because data URLs are, URLs and the `/`, `=` and `+` characters in base64 URLs may be encoded, so this needs a step calling `decodeURIComponent`.